src/ray/raylet/node_manager.cc

```diff
@@ -681,6 +742,34 @@ void NodeManager::ScheduleTasks() {
   }
 }
 
+void NodeManager::TreatTaskAsFailed(const TaskSpecification &spec) {
+  RAY_LOG(DEBUG) << "Treating task " << spec.TaskId() << " as failed.";
+  // Loop over the return IDs (except the dummy ID) and store a fake object in
+  // the object store.
+  int64_t num_returns = spec.NumReturns();
+  if (spec.IsActorTask()) {
+    // TODO(rkn): We subtract 1 to avoid the dummy ID. However, this leaks
+    // information about the TaskSpecification implementation.
+    num_returns -= 1;
```

Comment:
What happens if we also store the fake object for the dummy ID? I wonder if we could leverage that to automatically raise the exception for any following task on the same actor.

But I guess a concern here is memory leakage? Maybe it's okay since the actor is already dead.

Reply:
Well during normal execution we don't actually store dummy objects in the object store, so I don't think it makes sense to do that in the failure case.

To leverage that to automatically raise exceptions, I think we'd still have to recreate the actor (which is what we're trying to avoid with this code path).

Reply:
Okay, that makes sense.